src/NexusMods.DataModel/ArchiveMetaData/FilePathMetadata.cs

```diff
@@ -0,0 +1,16 @@
+﻿using NexusMods.DataModel.JsonConverters;
+using NexusMods.Paths;
+
+namespace NexusMods.DataModel.ArchiveMetaData;
+
+/// <summary>
+/// Archive metadata for a download that was installed from a file path.
+/// </summary>
+[JsonName("NexusMods.DataModel.ArchiveMetaData.FilePathMetadata")]
+public record FilePathMetadata : AArchiveMetaData
+{
+    /// <summary>
+    /// The filename portion of the path.
+    /// </summary>
+    public RelativePath OriginalName { get; init; } = RelativePath.Empty;
```

Comment:
There is a bug in the current code: new mods added with Add Mod button over the loadoutgrid have empty names:

![image](https://github.com/Nexus-Mods/NexusMods.App/assets/26797547/f11a7a87-be11-411a-8873-87fafc73275e)

The problem seems to be that for archives, the code is setting only `OriginalName` and not `AArchiveMetaData.Name` which is what is used in `archiveInstaller` to set the mod name. 

So either the init of OriginalName also sets `AArchiveMetaData.Name` or `Name` is initialized apart on each `RegisterDownload()` call throught the app, or `ArchiveInstaller` code is changed to check for `OriginalName` as well if `Name` is empty.

Reply:
nxm downloads have a name on the other hand, because `NxmDownloadProtocolHandler.cs` sets the defaultName parameter of AddMods using extracted mod name from the download:
https://github.com/Nexus-Mods/NexusMods.App/blob/6f4fd9ed003af1960f64111bd473392babc0c51f/src/NexusMods.CLI/Types/DownloadHandlers/NxmDownloadProtocolHandler.cs#L59

Which other places don't do.

Reply:
Installing mods from CLI without explicitly passing a name also results in mods without names being created.